src/anemoi/models/preprocessing/remapper.py

```diff
@@ -16,315 +16,58 @@
-        if x.shape[-1] == self.num_remapped_training_output_vars:
-            index = self.index_training_output
-            indices_remapped = self.index_training_backmapped_output
-            indices_keep = self.indices_keep_training_output
-            target_number_columns = self.num_training_output_vars
-
-        elif x.shape[-1] == self.num_remapped_inference_output_vars:
-            index = self.index_inference_output
-            indices_remapped = self.index_inference_backmapped_output
-            indices_keep = self.indices_keep_inference_output
-            target_number_columns = self.num_inference_output_vars
-
-        else:
-            raise ValueError(
-                f"Input tensor ({x.shape[-1]}) does not match the training "
-                f"({self.num_remapped_training_output_vars}) or inference shape ({self.num_remapped_inference_output_vars})",
-            )
-
-        # create new tensor with target number of columns
-        x_remapped = torch.zeros(x.shape[:-1] + (target_number_columns,), dtype=x.dtype, device=x.device)
-        if in_place and not self.printed_postprocessor_warning:
-            LOGGER.warning(
-                "Remapper (preprocessor) called with in_place=True. This preprocessor cannot be applied in_place as new columns are added to the tensors.",
-            )
-            self.printed_postprocessor_warning = True
-
-        # copy variables that are not remapped
-        x_remapped[..., indices_keep] = x[..., : len(indices_keep)]
-
-        # Backmap variables
-        for idx_dst, backmapper, idx_src in zip(index, self.backmappers, indices_remapped):
-            if idx_dst is not None:
-                x_remapped[..., idx_dst] = backmapper(x[..., idx_src])
-
-        return x_remapped
-
-
-class Remapper(BaseRemapperVariable):
-    """Remap and convert variables.
-
-    cos_sin:
-        Remap the variable to cosine and sine.
-        Map output back to degrees.
-
-    ```
-    cos_sin:
-      "mwd" : ["cos_mwd", "sin_mwd"]
-    ```
-    """
-
-    def __init__(
-        self,
-        config=None,
-        data_indices: Optional[IndexCollection] = None,
-        statistics: Optional[dict] = None,
-    ) -> None:
-        super().__init__(config, data_indices, statistics)
-
-        self.printed_preprocessor_warning, self.printed_postprocessor_warning = False, False
-
-        self._create_remapping_indices(statistics)
-
-        self._validate_indices()
+                raise ValueError(f"Unknown remapping method for {name}: {method}")
+
+    def transform(self, x, in_place: bool = True) -> torch.Tensor:
+        for name, method in self.methods.items():
+            idx = self.data_indices.data.input.name_to_index[name]
+            remapper = self.supported_methods[method][0]
+            x[..., idx] = remapper(x[..., idx])
+        return x
+
+    def inverse_transform(self, x, in_place: bool = True) -> torch.Tensor:
+        for name, method in self.methods.items():
```

Comment:
It could be that the index is `None` in case the input variable is forcing variable. Therefore, you need to check here, whether the index is `None`.